File: device-sdk/src/main/java/io/oddworks/device/metric/OddViewLoadMetric.java
package io.oddworks.device.metric;

public class OddViewLoadMetric extends OddMetric {
    private static final String TAG = OddViewLoadMetric.class.getSimpleName();

    protected static String viewLoadAction = ACTION_VIEW_LOAD;
    protected static boolean viewLoadEnabled = false;

    public OddViewLoadMetric() {

    }

    public static void setAction(String action) {
        viewLoadAction = action;
    }

    public static void setEnabled(boolean enabled) {
        viewLoadEnabled = enabled;
    }

    @Override
    public String getAction() {
        return viewLoadAction;
    }

    @Override
    public boolean getEnabled() {
        return viewLoadEnabled;
    }
}

File: device-sdk/src/main/java/io/oddworks/device/metric/OddViewLoadMetric.kt
package io.oddworks.device.metric

import org.json.JSONObject

class OddViewLoadMetric(contentType: String, contentId: String, title: String, meta: JSONObject? = null) : OddMetric(contentType, contentId, title, meta) {

    override val action: String
        get() = OddViewLoadMetric.action

    override val enabled: Boolean
        get() = OddViewLoadMetric.enabled

    companion object {
        private val TAG = OddViewLoadMetric::class.java.simpleName

        var action = OddMetric.ACTION_VIEW_LOAD
        var enabled = false
    }
}
